tools/flaky_test/process_xml.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+import subprocess
+import os
+import xml.etree.ElementTree as ET
+import slack
+import sys
+
+
+# Check if a test suite reports failure.
+def checkTestStatus(file):
+  tree = ET.parse(file)
+
+  root = tree.getroot()
+
+  for testsuite in root:
+    if (testsuite.attrib['failures'] != '0'):
+      return False
+  return True
+
+
+def parseXML(file, visited):
+  log_file = file.split('.')
+  log_file_path = ""
+
+  # This is dependent on the fact that log files reside in the same directory
+  # as their corresponding xml files.
+  for token in log_file[:-1]:
+    log_file_path += token
+  log_file_path += ".log"
+
+  tree = ET.parse(file)
+
+  root = tree.getroot()
+  ret = ""
+
+  # This loop is dependent on the structure of xml file emitted for test runs.
+  # Should this change in the future, appropriate adjustments need to made.
+  for testsuite in root:
+    if (testsuite.attrib['failures'] != '0'):
+      for testcase in testsuite:
+        for failure_msg in testcase:
+          if (testcase.attrib['name'], testsuite.attrib['name']) not in visited:
+            ret += "-----------------------Flaky Testcase: {} in TestSuite: {} -----------------------\n".format(
+                testcase.attrib['name'], testsuite.attrib['name'])
+            ret += log_file_path + "\n" + failure_msg.text + "\n"
+            visited.add((testcase.attrib['name'], testsuite.attrib['name']))
+  return ret
+
+
+def processFindOutput(f, problematic_files):
+  for line in f:
+    lineList = line.split('/')
+    filepath = ""
+    for i in range(len(lineList)):
+      if i >= len(lineList) - 2:
+        break
+      filepath += lineList[i] + "/"
+    filepath += "test.xml"
+    problematic_files[filepath] = line.strip('\n')
+
+
+# Prints out helpful information on the run using Git.
+# Should Git changes the output of the used commands in the future,
+# this will likely need adjustments as well.
+def getGitInfo(CI_TARGET):
+  ret = ""
+  os.system("git remote -v > ${TMP_OUTPUT_PROCESS_XML}")
+  os.system("git describe --all >> ${TMP_OUTPUT_PROCESS_XML}")
+  os.system("git show >> ${TMP_OUTPUT_PROCESS_XML}")
+  f = open(os.environ['TMP_OUTPUT_PROCESS_XML'], 'r+')
+  # The link should not change.
+  envoy_link = "https://github.com/envoyproxy/envoy"
```

Comment:
Don't hard code this. it should be readable from `os.environ['BUILD_REPOSITORY_URI']`

See https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml for what is available. You should be able to get azure pipelines URL to post as well.

Reply:
I see. I will take a look, thanks.